### PR TITLE
Optimize TestOrchestrator and add Start-Button

### PR DESCRIPTION
**Pull request by Morphclue**
# Description
This PR adjust the Schema in the `TestOrchestrator`. Also added a Start-Button to emit a Kafka-Signal.
Also refined the display of the Test-Environments into `ServiceName - CertificateName - STATUS`.

## Types of changes
- Enhancement

### CODE REVIEW

File: apps/srv/test-orchestrator-service/src/app/test-orchestrator/testOrchestrator.controller.ts [hunk 1]
```diff
@@ -58,4 +60,14 @@ export class TestOrchestratorController {
     await this.testOrchestratorService.deleteOne(testOrchestrator);
     return testOrchestrator;
   }
+
+  @Post('start/:id')
```
Comment (KosrenDQ) [inline apps/srv/test-orchestrator-service/src/app/test-orchestrator/testOrchestrator.controller.ts:64]: POST Methods should be on top of the controller, beside the other POST Methods.
The order matters.

File: apps/srv/test-orchestrator-service/src/app/test-orchestrator/testOrchestrator.schema.ts [hunk 2]
```diff
@@ -8,10 +8,10 @@ export class TestOrchestrator extends Document {
   @Prop({ required: true })
   status: string;
```
Comment (KosrenDQ) [inline apps/srv/test-orchestrator-service/src/app/test-orchestrator/testOrchestrator.schema.ts:10]: Shouldn't this still be required ?

File: libs/web/api/src/lib/test-environment.interface.ts [hunk 3]
```diff
@@ -1,6 +1,8 @@
 export interface TestEnvironment {
   readonly id?: string;
   readonly artifactId: string;
+  artifactName: string;
```
Comment (KosrenDQ) [inline libs/web/api/src/lib/test-environment.interface.ts:4]: readonly?

File: libs/web/api/src/lib/test-environment.service.ts [hunk 4]
```diff
@@ -28,4 +28,9 @@ export class TestEnvironmentService {
   deleteOne(id: string): Observable<TestEnvironment> {
     return this.http.delete<TestEnvironment>(`${this.environment.httpConf.environment}/${id}`);
   }
+
+  start(id: string, testDto: TestEnvironment): Observable<TestEnvironment> {
```
Comment (KosrenDQ) [inline libs/web/api/src/lib/test-environment.service.ts:32]: Mybe a more descriptive name ? :D

File: libs/web/test-environment/src/lib/edit/test-environment-edit.component.ts [hunk 5]
```diff
@@ -113,8 +113,10 @@ export class TestEnvironmentEditComponent {
 
     const environmentDto: TestEnvironment = {
```
Comment (KosrenDQ) [inline libs/web/test-environment/src/lib/edit/test-environment-edit.component.ts:114]: Why is there a name which is not setted after update?

File: libs/web/test-environment/src/lib/test-environment.component.ts [hunk 6]
```diff
@@ -4,7 +4,14 @@ import { Event, NavigationEnd, Router } from '@angular/router';
 import { Subscription } from 'rxjs';
```
Comment (KosrenDQ) [inline libs/web/test-environment/src/lib/test-environment.component.ts:4]: Need to talk about all changes

**Review by KosrenDQ — CHANGES_REQUESTED** [review]

File: apps/srv/test-orchestrator-service/src/app/test-orchestrator/testOrchestrator.controller.ts (hunk 1 above)
Comment (distrustME) [reply apps/srv/test-orchestrator-service/src/app/test-orchestrator/testOrchestrator.controller.ts]: Fixed in e212d0a.

File: apps/srv/test-orchestrator-service/src/app/test-orchestrator/testOrchestrator.schema.ts (hunk 2 above)
Comment (distrustME) [reply apps/srv/test-orchestrator-service/src/app/test-orchestrator/testOrchestrator.schema.ts:10]: Fixed in e212d0a.

File: libs/web/api/src/lib/test-environment.service.ts (hunk 4 above)
Comment (distrustME) [reply libs/web/api/src/lib/test-environment.service.ts]: Fixed in e212d0a.

File: libs/web/test-environment/src/lib/edit/test-environment-edit.component.ts (hunk 5 above)
Comment (distrustME) [reply libs/web/test-environment/src/lib/edit/test-environment-edit.component.ts]: Fixed in e212d0a.

File: libs/web/api/src/lib/test-environment.interface.ts (hunk 3 above)
Comment (distrustME) [reply libs/web/api/src/lib/test-environment.interface.ts:4]: Yes. The reason for that is, that the artifactName and certificateName will be later initialized inside the `test-environment.ts`. This is only to display information on the frontend. 
For this reason we implemented an additional TestEnvironmentDto inside `test-environment.interface.ts` to clarify the process now.

File: libs/web/test-environment/src/lib/test-environment.component.ts (hunk 6 above)
Comment (KosrenDQ) [reply libs/web/test-environment/src/lib/test-environment.component.ts:4]: Still the calls are not synchronized

**Review by KosrenDQ — CHANGES_REQUESTED** [review]

File: libs/web/test-environment/src/lib/test-environment.component.ts (hunk 6 above)
Comment (distrustME) [reply libs/web/test-environment/src/lib/test-environment.component.ts:4]: That's why i didn't wrote that it's fixed 👍 
Me and Fabian wanted to look into it today and try out the `concatMap()` Operator from `rxjs`.

Comment (distrustME) [reply libs/web/test-environment/src/lib/test-environment.component.ts]: Fixed in 362a88b and added more precise error-message in 3761f22.

Comment (distrustME) [reply libs/web/test-environment/src/lib/test-environment.component.ts]: Should be fixed in 8a935cb.

**Review by KosrenDQ — APPROVED** [review]

**Pull request merged**